rust/lance-index/benches/scalar_index.rs

```diff
@@ -0,0 +1,200 @@
+        Self::train_scalar_index(&index_store).await;
+
+        Self {
+            _datadir: tempdir,
+            index_store,
+            baseline_dataset,
+        }
+    }
+}
+
+async fn baseline_equality_search(fixture: &BenchmarkFixture) {
+    let mut stream = fixture
+        .baseline_dataset
+        .scan()
+        .filter("values == 10000")
+        .unwrap()
+        .with_row_id()
+        .try_into_stream()
+        .await
+        .unwrap();
+    let mut num_rows = 0;
+    while let Some(batch) = stream.try_next().await.unwrap() {
+        num_rows += batch.num_rows();
+    }
+    assert_eq!(num_rows, 1);
+}
+
+async fn warm_indexed_equality_search(index: &BTreeIndex) {
+    let row_ids = index
+        .search(&ScalarQuery::Equals(ScalarValue::UInt32(Some(10000))))
+        .await
+        .unwrap();
+    assert_eq!(row_ids.len(), 1);
+}
+
+async fn baseline_inequality_search(fixture: &BenchmarkFixture) {
+    let mut stream = fixture
+        .baseline_dataset
+        .scan()
+        .filter("values >= 50000000")
+        .unwrap()
+        .with_row_id()
+        .try_into_stream()
+        .await
+        .unwrap();
+    let mut num_rows = 0;
+    while let Some(batch) = stream.try_next().await.unwrap() {
+        num_rows += batch.num_rows();
+    }
+    // 100Mi - 50M = 54,857,600
+    assert_eq!(num_rows, 54857600);
+}
+
+async fn warm_indexed_inequality_search(index: &BTreeIndex) {
+    let row_ids = index
+        .search(&ScalarQuery::Range(
+            std::ops::Bound::Included(ScalarValue::UInt32(Some(50_000_000))),
+            std::ops::Bound::Unbounded,
+        ))
+        .await
+        .unwrap();
+    // 100Mi - 50M = 54,857,600
+    assert_eq!(row_ids.len(), 54857600);
+}
+
+async fn warm_indexed_isin_search(index: &BTreeIndex) {
+    let row_ids = index
+        .search(&ScalarQuery::IsIn(vec![
+            ScalarValue::UInt32(Some(10000)),
+            ScalarValue::UInt32(Some(50000000)),
+            ScalarValue::UInt32(Some(150000000)), // Not found
+            ScalarValue::UInt32(Some(287123)),
+        ]))
+        .await
+        .unwrap();
+    // Only 3 because 150M is not in dataset
+    assert_eq!(row_ids.len(), 3);
+}
+
+fn bench_baseline(c: &mut Criterion) {
+    let rt = tokio::runtime::Runtime::new().unwrap();
+
+    let fixture = rt.block_on(BenchmarkFixture::open());
+
+    c.bench_function("baseline_equality", |b| {
+        b.iter(|| rt.block_on(baseline_equality_search(&fixture)));
+    });
+
+    c.bench_function("baseline_inequality", |b| {
+        b.iter(|| rt.block_on(baseline_inequality_search(&fixture)));
+    });
+}
+
+fn bench_warm_indexed(c: &mut Criterion) {
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    let fixture = rt.block_on(BenchmarkFixture::open());
+
+    let index = rt
+        .block_on(BTreeIndex::load(fixture.index_store.clone()))
+        .unwrap();
+
+    c.bench_function("windexed_equality", |b| {
```

Comment:
I just keep on thinking of birds colliding into windows when I read this...

Reply:
"warm indexed" just didn't have the same ring to it :bird: :window: